File: shell-examples/src/main/js/io/vertx/example/shell/wget/wget_command.js
var CLI = require("vertx-js/cli");
var CommandBuilder = require("vertx-shell-js/command_builder");
var ShellService = require("vertx-shell-js/shell_service");
var CommandRegistry = require("vertx-shell-js/command_registry");

// Create the wget CLI
var cli = CLI.create("wget").setSummary("Wget implemented with Vert.x HTTP client").addArgument({
  "index" : 0,
  "argName" : "http-url",
  "description" : "the HTTP uri to get"
});

// Create the command
var helloWorld = CommandBuilder.command(cli).processHandler(function (process) {
  var url;
  try {
    url = new (Java.type("java.net.URL"))(process.commandLine().getArgumentValue(0));
  } catch(err) {
    process.write("Bad url\n").end();
    return
  }

  var client = process.vertx().createHttpClient();
  process.write("Connecting to " + url + "\n");
  var port = url.getPort();
  if (port === -1) {
    port = 80;
  }
  var req = client.get(port, url.getHost(), url.getPath());
  req.exceptionHandler(function (err) {
    process.write("wget: error " + err.getMessage() + "\n");
    process.end();
  });
  req.handler(function (resp) {
    process.write(resp.statusCode() + " " + resp.statusMessage() + "\n");
    var contentType = resp.getHeader("Content-Type");
    var contentLength = resp.getHeader("Content-Length");
    process.write("Length: " + (contentLength !== null ? contentLength : "unspecified"));
    if (contentType !== null) {
      process.write("[" + contentType + "]");
    }
    process.write("\n");
    process.end();
  });
  req.end();

}).build(vertx);

var service = ShellService.create(vertx, {
  "telnetOptions" : {
    "host" : "localhost",
    "port" : 3000
  }
});
CommandRegistry.getShared(vertx).registerCommand(helloWorld);
service.start(function (ar, ar_err) {
  if (!ar_err == null) {
    ar_err.printStackTrace();
  }
});

File: shell-examples/src/main/ruby/io/vertx/example/shell/wget/wget_command.rb
require 'vertx/cli'
require 'vertx-shell/command_builder'
require 'vertx-shell/shell_service'
require 'vertx-shell/command_registry'

# Create the wget CLI
cli = Vertx::CLI.create("wget").set_summary("Wget implemented with Vert.x HTTP client").add_argument({
  'index' => 0,
  'argName' => "http-url",
  'description' => "the HTTP uri to get"
})

# Create the command
helloWorld = VertxShell::CommandBuilder.command(cli).process_handler() { |process|

  begin
    url = Java::JavaNet::URL.new(process.command_line().get_argument_value(0))
  rescue
    process.write("Bad url\n").end()
    return
  end

  client = process.vertx().create_http_client()
  process.write("Connecting to #{url}\n")
  port = url.get_port()
  if (port == -1)
    port = 80
  end
  req = client.get(port, url.get_host(), url.get_path())
  req.exception_handler() { |err|
    process.write("wget: error #{err.get_message()}\n")
    process.end()
  }
  req.handler() { |resp|
    process.write("#{resp.status_code()} #{resp.status_message()}\n")
    contentType = resp.get_header("Content-Type")
    contentLength = resp.get_header("Content-Length")
    process.write("Length: #{(contentLength != nil ? contentLength : "unspecified")}")
    if (contentType != nil)
      process.write("[#{contentType}]")
    end
    process.write("\n")
    process.end()
  }
  req.end()

}.build($vertx)

service = VertxShell::ShellService.create($vertx, {
  'telnetOptions' => {
    'host' => "localhost",
    'port' => 3000
  }
})
VertxShell::CommandRegistry.get_shared($vertx).register_command(helloWorld)
service.start() { |ar_err,ar|
  if (!ar_err == nil)
    ar_err.print_stack_trace()
  end
}
